File: trustloop-desktop/src/main/secrets.ts
import { SecretsManagerClient, GetSecretValueCommand } from "@aws-sdk/client-secrets-manager";

const SECRET_NAME = process.env.AWS_SECRET_NAME || "trustloop/production";
const REGION = process.env.AWS_REGION || "us-east-1";

/**
 * In production, fetch ALL env vars from AWS Secrets Manager —
 * the same JSON blob the web app's ECS tasks use (trustloop/production).
 * In development, .env file is sufficient — this is a no-op.
 */
export async function loadAwsSecrets(): Promise<void> {
  if (process.env.NODE_ENV !== "production") return;

  const client = new SecretsManagerClient({ region: REGION });
  const res = await client.send(new GetSecretValueCommand({ SecretId: SECRET_NAME }));
  if (!res.SecretString) throw new Error(`Empty secret: ${SECRET_NAME}`);

  const secrets: Record<string, string> = JSON.parse(res.SecretString);
  for (const [key, value] of Object.entries(secrets)) {
    // Don't overwrite values explicitly set via CLI or environment
    if (!process.env[key]) process.env[key] = value;
  }
}

File: trustloop-desktop-tauri/src-tauri/src/secrets.rs
/// In production, fetch ALL env vars from AWS Secrets Manager —
/// the same JSON blob the web app's ECS tasks use (trustloop/production).
/// In development, .env file is sufficient — this is a no-op.
pub async fn load_aws_secrets() -> Result<(), String> {
    if std::env::var("NODE_ENV").unwrap_or_default() != "production" {
        return Ok(());
    }

    let secret_name = std::env::var("AWS_SECRET_NAME").unwrap_or_else(|_| "trustloop/production".into());
    let region = std::env::var("AWS_REGION").unwrap_or_else(|_| "us-east-1".into());

    let config = aws_config::defaults(aws_config::BehaviorVersion::latest())
        .region(aws_config::Region::new(region))
        .load()
        .await;

    let client = aws_sdk_secretsmanager::Client::new(&config);
    let resp = client
        .get_secret_value()
        .secret_id(&secret_name)
        .send()
        .await
        .map_err(|e| format!("AWS Secrets Manager error: {}", e))?;

    let secret_string = resp.secret_string().ok_or("Empty secret")?;
    let secrets: std::collections::HashMap<String, String> =
        serde_json::from_str(secret_string).map_err(|e| format!("Parse error: {}", e))?;

    for (key, value) in secrets {
        // Don't overwrite values explicitly set via CLI or environment
        if std::env::var(&key).is_err() {
            std::env::set_var(&key, &value);
        }
    }

    Ok(())
}
